### PR TITLE
Add startsWith for string filter

### PR DESCRIPTION
**Pull request by ssanjay1**
` fullName: { $startsWith: "Hemsworth" },`

### CODE REVIEW

File: examples/basic/sdk/ontology.json [hunk 1]
```diff
@@ -156,6 +156,11 @@
               "type": "string"
             }
           },
+          "name": {
+            "dataType": {
+              "type": "string"
+            }
+          },
```
Comment (mfedderly) [inline examples/basic/sdk/ontology.json:163]: is this logically a duplicate of `usState`?

File: packages/client/src/query/WhereClause.ts [hunk 2]
```diff
@@ -32,7 +32,8 @@ export type PossibleWhereClauseFilters =
   | "$lt"
   | "$lte"
   | "$within"
-  | "$intersects";
+  | "$intersects"
+  | "$startsWith";
```
Comment (mfedderly) [inline packages/client/src/query/WhereClause.ts:36]: Its kind of wild that we didn't need to write any additional converter code for this to work.

**Review by mfedderly — APPROVED** [review]

File: examples/basic/sdk/ontology.json (hunk 1 above)
Comment (ssanjay1) [reply examples/basic/sdk/ontology.json:163]: I guess yea logically, usState is the two letter abbreviation though and I wanted to make sure it worked with strings that had spaces in it. I can remove though

File: packages/client/src/query/WhereClause.ts (hunk 2 above)
Comment (ssanjay1) [reply packages/client/src/query/WhereClause.ts:36]: Yea I was surprised too

**Pull request merged**